packages/rum-core/src/browser/domMutationObservable.ts

```diff
@@ -1,8 +1,64 @@
-import { monitor } from '@datadog/browser-core'
-import { LifeCycle, LifeCycleEventType } from '../domain/lifeCycle'
+import { monitor, Subscription } from '@datadog/browser-core'
 
-type MutationObserverConstructor = new (callback: MutationCallback) => MutationObserver
+export interface DOMMutation {
+  subscribe(callback: () => void): Subscription
+}
+
+export function createDOMMutationObservable(): DOMMutation {
+  let callbacks: Array<() => void> = []
+  const MutationObserver = getMutationObserverConstructor()
+  const observer = MutationObserver ? new MutationObserver(monitor(notify)) : undefined
+  let isDOMObserved = false
+
+  function notify() {
+    callbacks.forEach((callback) => callback())
+  }
 
+  function startDOMObservation() {
+    if (!observer) {
+      return
+    }
+    observer.observe(document, {
+      attributeOldValue: true,
+      attributes: true,
+      characterData: true,
+      characterDataOldValue: true,
+      childList: true,
+      subtree: true,
```

Comment:
Any reason to change the options here? I think we don't need the `*OldValue` ones

Reply:
You right